src/smc-webapp/jupyter/top-buttonbar.tsx

```diff
@@ -187,39 +197,53 @@ export class TopButtonbar0 extends Component<TopButtonbarProps> {
     ]);
   }
 
-  cell_select_type(event: any): void {
-    this.props.frame_actions.set_selected_cell_type(event.target.value);
+  cell_select_type(key: any): void {
+    this.props.frame_actions.set_selected_cell_type(key);
     this.focus();
   }
 
-  render_select_cell_type() {
-    let cell_type: any;
-    if (this.props.sel_ids.size > 1) {
-      cell_type = "multi";
-    } else {
-      cell_type = this.props.cells.getIn(
-        [this.props.cur_id, "cell_type"],
-        "code"
-      );
+  private static cell_type_title(key): string {
```

Comment:
Is there a reason this is a private static member instead of a function off the component?

Reply:
that way it's in close proximity of where it is used – better for dev experience, that's all

Reply:
I don't understand @haraldschilly 's explanation.